### ISSUE TITLE
Extending the theme to a new variant

### ISSUE THREAD

**Issue by egotch** [barrientosvctor/abyss.nvim#49]
Hello!

I've been a big fan of the Abyss theme for well over a year now, and I absolutely love the changes you've brought to this and the active maintenance you've kept up with!  There's a lot to love with the abyss them, and those blues/pinks are what make this one of the most chill themes in my opinion.

I'd like the opportunity to contribute to this project with a new variant on the color scheme.  Specifically I think having a variant of the theme using a palette based on an aurora/northern lights would be beautiful.

I had tried opening up a branch to get this variant started, but it looks like I don't have permissions to do so.  What would be needed for me to contribute to this project with a new working branch?

**Comment by barrientosvctor** [comment]
Hello @egotch, thanks for your motivation for contributing to this project! I really like your proposal.

To fix your problem for branch creating, fork this repo and make your changes in a new branch based on `develop` branch for then create a pull request. Remember to follow the [contribution rules](https://github.com/barrientosvctor/abyss.nvim?tab=readme-ov-file#-contributing).

**Comment by barrientosvctor** [comment]
Hello @egotch, I'm realizing that making a variant for Abyss is necessary
to create a kind of standardized structure for color palette-making, this
for maintainable code purposes. May you mind letting me first create it on
the `develop` branch and then you continue making the color palette for the
new variant?

**Comment by egotch** [comment]
@barrientosvctor ,

Yes that should be just fine.  It shouldn't be hard to merge my changes into whatever the new structure is. Let me know when that part is set. No rush 🙂

**Comment by egotch** [comment]
@barrientosvctor:

Just wanted to give you a preview of the tinkering I've been doing.  I still have a fair amount to go on the specifics for functions, keywords, brackets, and all. 

But you can start to get a sense of the color pallette coming  together here with the cool purples, pinks, and teals:
<img width="889" alt="preview1" src="https://github.com/user-attachments/assets/9be4cff9-dea6-4da4-bb88-9e00c9981dcd" />
<img width="908" alt="Preview2" src="https://github.com/user-attachments/assets/6b2ca8ce-11ea-4749-ac02-51272cf2b8bb" />
<img width="1280" alt="Preview3_go_bg" src="https://github.com/user-attachments/assets/101571ea-233d-453f-8557-685d69f58635" />
<img width="1280" alt="Preview3_go_transbg" src="https://github.com/user-attachments/assets/2eef7a87-31a7-4063-b1f3-e800457fc91a" />

**Comment by barrientosvctor** [comment]
Sorry for replying late, @egotch. I just finished the changes for color palettes-making, they're on `develop` branch. I don't know how much my changes affects your color palette, any suggestion for these changes or if something goes wrong, you can tell me.

Soon I'll commit the test cases for color palettes-making.

**Comment by egotch** [comment]
@barrientosvctor - love the changes.  This looks a lot like how nightfox and the other big themes are structured.  This will be super easy to build out and test the additional palette now.  Thanks for getting that together!

**Comment by barrientosvctor** [comment]
Glad you like it


All you need to do for creating a new variant is just create its file on
'palettes' folder, naming it as you want the new variant would be named.
You can follow the way how 'abyss' palette was created to create your own.

**Comment by barrientosvctor** [comment]
@egotch 

I just committed the test cases file for the creation of new variants. You can start making it.

**Comment by egotch** [comment]
@barrientosvctor sounds good I'll get that pulled, and, just to confirm, to test the pallet, should I be able to run a vim colorscheme command and see it as an option?  Or do I need to update a .vim file on the extension.

**Comment by barrientosvctor** [comment]
To see your color scheme variant as a option to select when typing the
`:colorscheme` vim command, you need to create a new vim script file on
`colors/*.vim`. Name the file as the name of your variant.

**Comment by egotch** [comment]
Perfect. Makes sense. I'll likely have the first cut at the variant next week after the holidays. Thanks for making the updates to get this going easier!

**Comment by barrientosvctor** [comment]
Glad to help, take the time you need to finish the variant. Have a good
holiday!

**Comment by egotch** [comment]
@barrientosvctor - sending in the PR.  Let me know what you think.  I added a little bit of granularity to the theme, and added a couple of changes to the original abyss theme to keep them up to date

**Comment by egotch** [comment]
ehhh, scratch that.  I'll make some changes to the readme as well to give previews of the new theme

**Comment by barrientosvctor** [comment]
> ehhh, scratch that. I'll make some changes to the readme as well to give previews of the new theme

Ok no problem, I'll be waiting for your changes.

**Comment by egotch** [comment]
> > ehhh, scratch that. I'll make some changes to the readme as well to give previews of the new theme
> 
> Ok no problem, I'll be waiting for your changes.

Should be sending it in tonight.  I've been using the new theme exclusively this week for work and it's been operating just great with all of the extensions.

**Comment by barrientosvctor** [comment]
> > > ehhh, scratch that. I'll make some changes to the readme as well to give previews of the new theme
> > 
> > 
> > Ok no problem, I'll be waiting for your changes.
> 
> Should be sending it in tonight. I've been using the new theme exclusively this week for work and it's been operating just great with all of the extensions.

It sounds excellent!

**Comment by egotch** [comment]
@barrientosvctor  PR is in.  Let me know what you think and if there are any issues/changes.

**Comment by barrientosvctor** [comment]
> @barrientosvctor PR is in. Let me know what you think and if there are any issues/changes.

I'm glad for your contribution, thank you so much!

**Comment by egotch** [comment]
> > @barrientosvctor PR is in. Let me know what you think and if there are any issues/changes.
> 
> I'm glad for your contribution, thank you so much!

No problem!

I've loved this theme since my vs code says. Happy to off continuing support and contributions as I come across them in the new variant if you're ok with that!

Also...thoughts on a tmux theme?  I was potentially going to look into that next, those catpuccin folks don't have to be the only ones with cool consistency!

**Comment by barrientosvctor** [comment]
> Happy to off continuing support and contributions as I come across them in the new variant if you're ok with that!

@egotch Sure you can! In fact, I'll continue making updates to the theme core and I'm not planning to make changes in the Abyss boreal variant or any file related to it (beyond than updating the spec when new properties are created) because you're the first creator/maintainer of whose variant. Feel free to contribute if you consider needed.

> Also...thoughts on a tmux theme? I was potentially going to look into that next, those catpuccin folks don't have to be the only ones with cool consistency!

I never thought about that, It'd be great for implementing to Abyss.
